platform/external-system-impl/src/com/intellij/openapi/externalSystem/service/execution/ExternalSystemProcessHandler.java

```diff
@@ -69,7 +69,7 @@ public void notifyTextAvailable(@NotNull final String text, @NotNull final Key o
   }
 
   @Override
-  protected void destroyProcessImpl() {
+  protected synchronized void destroyProcessImpl() {
```

Comment:
Instead of the synchronized keywords, Radomir Malaczek suggests a local variable (Task task = myTask) here and in closeInput, in case you prefer that approach.